### PR TITLE
feat(twap): integrate TradeConfirmModal in Twap orders

### LINKED ISSUE #2429: Create a generic component TradeConfirmationModal (to replace LimitOrdersConfirmModal)
image

For Advanced orders, we will need a component for the confirmation modal.
Need to refactor `LimitOrdersConfirmModal` to create a new universal solution for confirming any trade.

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #2429

This is a continuation of https://github.com/cowprotocol/cowswap/pull/2573.
Current PR implements only basic features of `TradeConfirmModal`. There are some issues with TODO:
 - details displaying, like a limit price, num of part, etc.
 - price impact displaying
 - the link on TWAP order creation tx is wrong (it leads to Etherscan instead of Safe)
 - confirmation warnings

All these will be fixed in the next iterations.

  # To Test

1. Twap order creation should go through the confirmation modal

### CODE REVIEW

File: src/modules/tradeQuote/hooks/useTradeQuotePolling.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { onlyResolvesLast } from 'legacy/utils/async'
 import { useUpdateCurrencyAmount } from 'modules/trade/hooks/useUpdateCurrencyAmount'
 import { updateTradeQuoteAtom } from 'modules/tradeQuote/state/tradeQuoteAtom'
 
-import { getQuote } from 'api/gnosisProtocol'
+import { getQuote } from 'api/gnosisProtocol/api'
```
Comment (shoom3301) [inline src/modules/tradeQuote/hooks/useTradeQuotePolling.ts:11]: There was a circular dependency error

File: src/modules/twap/hooks/useCreateTwapOrder.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import { settleTwapOrder } from '../services/settleTwapOrder'
+import { twapOrderAtom } from '../state/twapOrderAtom'
+
+export function useCreateTwapOrder() {
+  const twapOrder = useAtomValue(twapOrderAtom)
+
+  const { inputCurrencyAmount, outputCurrencyAmount } = useAdvancedOrdersDerivedState()
+
+  const twapOrderCreationContext = useTwapOrderCreationContext(inputCurrencyAmount as Nullish<CurrencyAmount<Token>>)
```
Comment (alfetopito) [inline src/modules/twap/hooks/useCreateTwapOrder.ts:22]: What type does `inputCurrencyAmount` come as from `useAdvancedOrdersDerivedState`?

Cab we maybe update `useAdvancedOrdersDerivedState` or `useTwapOrderCreationContext` to work with the same type?

**Review by alfetopito — COMMENTED** [review]
Code looks mostly ok.
Checked the confirmation modal on Cosmos and looks good.

Code not test it in the app though.
When I loaded on my Safe, I get stuck on setting the fallback handler.
image

Should this be working already or not?
If not, what can I do?

Also, can you add test instructions for getting there?

**Comment by shoom3301** [comment]
@alfetopito currently, TWAP works only on Goerli. Sorry, I forgot to mention that

File: src/modules/twap/hooks/useCreateTwapOrder.ts (hunk 2 above)
Comment (shoom3301) [reply src/modules/twap/hooks/useCreateTwapOrder.ts:22]: `<CurrencyAmount<Currency>> !== <CurrencyAmount<Token>>` unfortunately.
The only difference is `Currency` can be `NativeCurrency | Token`, but in advanced order we don't use `NativeCurrency`

Comment (alfetopito) [reply src/modules/twap/hooks/useCreateTwapOrder.ts:22]: Ah got it!

**Review by alfetopito — APPROVED** [review]

**Pull request merged**